crates/catalog/rest/src/catalog.rs

```diff
@@ -130,6 +130,33 @@ impl RestCatalogConfig {
             );
         }
 
+        for (key, value) in self.props.iter() {
+            if let Some(stripped_key) = key.strip_prefix("header.") {
+                headers.insert(
+                    HeaderName::from_str(stripped_key).map_err(|e| {
+                        Error::new(
+                            ErrorKind::DataInvalid,
+                            format!("Invalid header name: {stripped_key}!"),
+                        )
+                        .with_source(e)
+                    })?,
+                    HeaderValue::from_str(value).map_err(|e| {
+                        Error::new(
+                            ErrorKind::DataInvalid,
+                            format!("Invalid header value: {value}!"),
+                        )
+                        .with_source(e)
+                    })?,
+                );
+            }
+        }
+        Ok(headers)
+    }
+
+    fn try_create_rest_client(&self) -> Result<HttpClient> {
+        // TODO: We will add ssl config, sigv4 later
+        let headers = self.get_default_headers()?;
```

Comment:
Refactor code into a new function `get_default_headers` for easier unit testing.